src/server/status_server.rs

```diff
@@ -89,11 +108,92 @@ impl StatusServer {
     }
 }
 
+// For handling fail points related requests
+#[cfg(not(feature = "no-fail"))]
+fn handle_fail_points_request(
+    req: Request<Body>,
+) -> Box<dyn Future<Item = Response<Body>, Error = hyper::Error> + Send> {
+    let path = req.uri().path().to_owned();
+    let method = req.method().to_owned();
+    let fail_path = format!("{}/", FAIL_POINTS_REQUEST_PATH);
+    let fail_path_has_sub_path: bool = path.starts_with(&fail_path);
+
+    match (method, fail_path_has_sub_path) {
+        (Method::PUT, true) => Box::new(req.into_body().concat2().map(move |chunk| {
+            let (_, name) = path.split_at(fail_path.len());
+            if name.is_empty() {
+                return Response::builder()
+                    .status(StatusCode::UNPROCESSABLE_ENTITY)
```

Comment:
Looks like 400 BAD REQUEST is a better error code?

Reply:
Maybe, I basically borrowed the same error status from the hyper example code:
https://github.com/hyperium/hyper/blob/master/examples/params.rs

Just let me know again if you think I should change it.